File: benchmarks/langages1/count/nn.cpp
#include <iostream>
#include <fstream>
#include <chrono>

int main() {
    std::ifstream file("apps/count/n.txt");
    long long n;
    file >> n;

    auto start = std::chrono::steady_clock::now();
    volatile long long count = 0;
    for (long long i = 0; i < n; ++i) {
        count++;
    }
    auto end = std::chrono::steady_clock::now();
    double duration = std::chrono::duration<double>(end - start).count();

    std::cout << "✅ C++ → Counted to " << count << " in " << duration << "s\n";
}

File: benchmarks/langages1/count/nn.rs
use std::fs::read_to_string;
use std::time::Instant;

fn main() {
    let n: u64 = read_to_string("apps/count/n.txt").unwrap().trim().parse().unwrap();

    let start = Instant::now();
    let mut count = 0;
    for _ in 0..n {
        count += 1;
    }
    let duration = start.elapsed().as_secs_f64();

    println!("✅ Rust → Counted to {} in {:.3}s", count, duration);
}
